nevergrad/functions/multiobjective/core.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Tuple, Any, Callable, List, Dict
+import numpy as np
+from nevergrad.common.typetools import ArrayLike
+from .pyhv import _HyperVolume
+
+
+ArgsKwargs = Tuple[Tuple[Any, ...], Dict[str, Any]]
+
+
+class MultiobjectiveFunction:
+    """Given several functions, and threshold on their values (above which solutions are pointless),
+    this function returns a single-objective function, correctly instrumented, the minimization of which
+    yields a solution to the original multiobjective problem.
+
+    multi_objective_function: objective functions, to be minimized, of the original multiobjective problem.
+    upper_bounds: upper_bounds[i] is a threshold above which x is pointless if functions[i](x) > upper_bounds[i].
+
+    Returns an objective function to be minimized (it is a single objective function).
+    Warning: this function is not stationary.
+    The minimum value obtained for this objective function is -h,
+    where h is the hypervolume of the Pareto front obtained, given upper_bounds as a reference point.
+    """
+
+    def __init__(self, multiobjective_function: Callable[..., ArrayLike], upper_bounds: ArrayLike) -> None:
+        self.multiobjective_function = multiobjective_function
+        self._upper_bounds = np.array(upper_bounds, copy=False)
+        self._hypervolume: Any = _HyperVolume(self._upper_bounds)  # type: ignore
+        self._points: List[Tuple[ArgsKwargs, np.ndarray]] = []
+        self._best_volume = -float("Inf")
+
+    def compute_aggregate_loss(self, losses: ArrayLike, *args: Any, **kwargs: Any) -> float:
+        # We compute the hypervolume
+        arr_losses = np.minimum(np.array(losses, copy=False), self._upper_bounds)
```

Comment:
wouldnt it help if we return something like `np.max(np.maximum(np.array(losses, copy=False) - self._upper_bounds, 0))` (i.e. the maximum part above the bounds) when it is strictly positive?

Reply:
yes! done.